src/Apps/Order/Components/AddressVerificationFlow.tsx

```diff
@@ -166,13 +244,30 @@ const AddressVerificationFlow: React.FC<AddressVerificationFlowProps> = ({
         </RadioGroup>
         <Spacer y={4} />
         <Flex width="100%" justifyContent="space-between">
-          <Button onClick={handleClose} variant="secondaryBlack" flex={1}>
+          <Button onClick={handleBackToEdit} variant="secondaryBlack" flex={1}>
             Back to Edit
           </Button>
           <Spacer x={1} />
           <Button
             disabled={!(selectedAddressKey && selectedAddressKey.length > 0)}
-            onClick={chooseAddress}
+            onClick={() => {
+              if (selectedAddressKey) {
+                trackEvent({
+                  action_type: "clickedValidationAddress",
+                  context_module: ContextModule.ordersShipping,
+                  context_page_owner_type: OwnerType.ordersShipping,
+                  context_page_owner_id: contextPageOwnerSlug,
+                  user_id: user?.id,
+                  subject: "Confirm your delivery address",
+                  option: selectedAddressKey.includes("suggestedAddress")
+                    ? "Recommended"
+                    : "What you entered",
+                  label: "Use This Address",
+                })
+              }
+
+              chooseAddress()
```

Comment:
I see inconsistent order of `chooseAddress` and `trackEvent` between here and the `REVIEW_AND_CONFIRM` below. I assume we'll want to leave tracking at the last step of the handling?

Reply:
My preference has been to `trackEvent` before triggering the event handler (`chooseAddress`, `onClose`, etc.) so that we can capture the user's intention before performing an action that has side-effects. I am worried about the case where that action errors-out or the side-effects cause the tracking event to not be fired.

I don't feel strongly about this preference as long as we are consistent, so I agree that the call order of one or the other should be updated.

Reply:
I'm going to maintain the order that @iskounen is talking about here to be able to capture the user intent one line of code before the user action is taken for all tracking calls in this PR before merging. This can be easily reversible if not an ideal setup also, and I can QA on staging to make sure. 

IMO it does make more sense to track AFTER use action, but aren't there also scenarios where this can lead to inconsistent if the user action is inconsistent?? 

However, @damassi or @dzucconi wondering if either of you have preferences on where in the logic flow tracking calls should be placed in Force, before OR after user actions?? Any input on this?

Reply:
I've used enough checkouts with ad blockers installed to understand that tracking calls should always happen after any logic executes. In our case this shouldn't matter but to me it makes sense that you'd only want to track something once it happened.

Reply:
For something like a click event I would do something like this:

```ts
<Button onClick={() => {
  tracking.trackEvent(...)
  
  doSomethingElse()
}}>Click Me</Button>
```

Then we can be sure that the tracking event fired before anything else happened. I wouldn't overthink it too much or code too defensively here. If something fails in `doSomethingElse` it could mess up the stats, but that's a bug and hopefully it would be caught and addressed quickly and averaged out as a deviation.

Reply:
I've got tracking being called directly after button clicks and when onClose() handlers are being called exactly like how you have it described @damassi, but to @dzucconi's point it's important to make sure to track an action after it was taken regardless. Thank you for the input!

Reply:
Ah yes, @dzucconi does make a very good point about ad blockers etc. I'm sure its all good here tho